esx_service/utils/vmdk_utils.py

```diff
@@ -67,20 +78,24 @@ def get_volumes():
 
 def get_vmdk_path(path, vol_name):
     """If the volume-related VMDK exists, returns full path to the latest
-    VMDK disk in the disk chain, be it volume-xxxxx.vmdk or volume.vmdk.
+    VMDK disk in the disk chain, be it volume-NNNNNN.vmdk or volume.vmdk.
     If the disk does not exists, returns full path to the disk for create().
     """
 
-    # Get a delta disk list, and if it's empty - return the full path for volume VMDK base file
-    # Note: we rely on NEVER allowing '-' in volume name and on the fact that ESXi
-    # always creates deltadisks as <name>-DDDDDD.vmdk (D is a digit) for delta disks
-    delta_disks = glob.glob("{0}/{1}-*[0-9].vmdk".format(path, vol_name))
+    # Get a delta disk list, and if it's empty - return the full path for volume
+    # VMDK base file.
+    # Note: we rely on NEVER allowing '-NNNNNN' in end of a volume name and on
+    # the fact that ESXi always creates deltadisks as <name>-NNNNNN.vmdk (N is a
+    # digit, and there are exactly 6 digits there) for delta disks
+    #
+    # see vmdk_ops.py:parse_vol_name() which enforces the volume name rules.
+    delta_disks = glob.glob("{0}/{1}{2}".format(path, vol_name, SNAP_SUFFIX_GLOB))
     if not delta_disks:
         return os.path.join(path, "{0}.vmdk".format(vol_name))
 
-    # this funky code gets the name of the youngest delta disk:
+    # this funky code gets the name of the latest delta disk:
     latest = sorted([(vmdk, os.stat(vmdk).st_ctime) for vmdk in delta_disks], key=lambda d: d[1], reverse=True)[0][0]
-    logging.debug("The youngest delta disk is %s. All delta disks: %s", latest, delta_disks)
+    logging.debug("The latest delta disk is %s. All delta disks: %s", latest, delta_disks)
     return latest
```

Comment:
This may be the newest delta disk created but its not necessarily the delta disk in use, if the user is moving between snaps. In which case the user can be on a delta-disk thats not the newest (by ctime) but in use. Wonder if we should really store the bus and unit number at which the disk is connected to the PVSCSI controller in the KV and when we get a request we only query the VM config to get the disk thats at that location. Then we get the exact disk thats in use no matter which snao the VM is running on.